torch/nn/modules/module.py

```diff
@@ -595,19 +595,36 @@ def to(self, *args, **kwargs):
             tensor([[ 0.1914, -0.3420],
                     [-0.5112, -0.2324]], dtype=torch.float16)
 
+            >>> linear = nn.Linear(2, 2, bias=None).to(torch.cdouble)
+            >>> linear.weight
+            Parameter containing:
+            tensor([[ 0.3741+0.j,  0.2382+0.j],
```

Comment:
`weight` is initialized using [kaiming_uniform_](https://pytorch.org/docs/stable/nn.init.html#torch.nn.init.kaiming_uniform_). Yes, if `Linear` had a `dtype` argument, then it could initialize complex-valued weights.